### PR TITLE
Test of loongson with binutils_2.43-gcc_14.2.0-glibc_2.40

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

File: docker/cross-files/loongarch64-gcc-14-ccache.cross [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[binaries]
+c = ['ccache', 'loongarch64-unknown-linux-gnu-gcc']
+cpp = ['ccache', 'loongarch64-unknown-linux-gnu-g++']
+ar = 'loongarch64-unknown-linux-gnu-ar'
+strip = 'loongarch64-unknown-linux-gnu-strip'
+objcopy = 'loongarch64-unknown-linux-gnu-objcopy'
+ld = 'loongarch64-unknown-linux-gnu-ld'
+exe_wrapper = ['qemu-loongarch64-static', '-L', '/opt/cross-tools/target/usr']
+
+[properties]
```
Comment (yinshiyou) [inline docker/cross-files/loongarch64-gcc-14-ccache.cross:10]: Should add '-mlsx' to compile LSX builtin code.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,891 +15,43 @@ concurrency:
...
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
     - uses: actions/checkout@v4
     - name: CPU Information
       run: cat /proc/cpuinfo
     - name: Install APT Dependencies
       run: |
-        sudo add-apt-repository ppa:savoury1/virtualisation
         sudo apt-get install -y ninja-build ninja-build meson qemu-user-static
```
Comment (yinshiyou) [inline .github/workflows/ci.yml:27]: I will try to confirm if this qemu support LSX.

File: docker/cross-files/loongarch64-gcc-14-ccache.cross (hunk 1 above)
Comment (mr-c) [reply docker/cross-files/loongarch64-gcc-14-ccache.cross:10]: Is that a feature of all loongson CPUs?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mr-c) [reply .github/workflows/ci.yml:27]: It should, https://packages.ubuntu.com/oracular/qemu-user-static is version `1:9.0.2+ds-4ubuntu3`

Checking by running the container locally I get
```
root@27baec51c897:/# qemu-loongarch64-static -cpu help
Available CPUs:
  la132
  la464
  max
root@27baec51c897:/# qemu-loongarch64-static -version
qemu-loongarch64 version 9.0.2 (Debian 1:9.0.2+ds-4ubuntu3)
Copyright (c) 2003-2024 Fabrice Bellard and the QEMU Project developers
```

Comment (mr-c) [reply .github/workflows/ci.yml:27]: Yep, the CI passed with `-cpu la464` to qemu against `-mlsx` with gcc

https://github.com/simd-everywhere/simde/actions/runs/10809969904/job/29986199744?pr=1217

**Comment by mr-c** [comment]
Great the build even works with the loongson gcc-14 from Ubuntu, so lets use that. One more test, I will revert your fixes to confirm that they are needed.

**Comment by mr-c** [comment]
Excellent, the expected failure occured! https://github.com/simd-everywhere/simde/actions/runs/10810965402/job/29989335235?pr=1217

**Pull request closed without merging**